tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -577,6 +578,28 @@ public:
    void CommitCluster() final;
 };
 
+/// The generic field for `std::pair<T1, T2>` types
+class RPairField : public RRecordField {
+private:
+   TClass *fClass{};
+   static std::string GetTypeList(const std::vector<std::unique_ptr<Detail::RFieldBase>> &itemFields);
+
+protected:
+   std::unique_ptr<Detail::RFieldBase> CloneImpl(std::string_view newName) const override;
+
+   RPairField(std::string_view fieldName, std::vector<std::unique_ptr<Detail::RFieldBase>> &&itemFields,
```

Comment:
How about using and `std::pair` for the `itemFields` and `offsets` parameters. Also in the other constructor. We avoid then the need to check the vector lengths.

Reply:
The problem with that is on the call to the `RRecordField` delegated constructor: it takes a `std::vector<std::unique_ptr<Detail::RFieldBase>>`.  In that case, we cannot use aggregate initialization, i.e. `{ ... }` to initialize the `std::vector` from the elements in the `std::pair` as `std::initializer_list` is always passed by copy and `std::unique_ptr` is not copiable.

Of course, we could pass an empty initializer list and add item fields in the body of the constructor, but I find that less elegant, so I vote for leaving it as-is.